src/main/java/com/softserveinc/task02/Task02.java

```diff
@@ -20,32 +23,92 @@
 
     static List<Employee> ex01() {
         // TODO: find and return list of all male employee with aged 18 to 27 (inclusive)
-        return null;
+
+//        class MalePredicate implements Predicate<Employee>{
+//
+//            @Override
+//            public boolean test(Employee employee) {
+//                return employee.getGender() == Gender.MAN;
+//            }
+//        }
+//
+//
+//        Predicate<Employee> agePredicate = new Predicate<>() {
+//            @Override
+//            public boolean test(Employee employee) {
+//                return employee.getAge() > 18 && employee.getAge() <= 27;
+//            }
+//        };
+//        MalePredicate malePredicate = new MalePredicate();
+//        List<Employee> male = new ArrayList<>();
+//        for (Employee employee: EMPLOYEES){
+//            if (malePredicate.test(employee)){
+//                male.add(employee);
+//            }
+//        }
+//
+//        List<Employee> maleAged18to27 = new ArrayList<>();
+//        for (Employee employee: male) {
+//            if (agePredicate.test(employee)){
+//                maleAged18to27.add(employee);
+//            }
+//        }
+//        return maleAged18to27;
+
+        return EMPLOYEES
+                .stream()
+                .filter(employee -> employee.getGender() == Gender.MAN)
+                .filter(man -> man.getAge() >= 18 && man.getAge() <= 27)
+                .collect(Collectors.toList());
+
+
     }
 
     static double ex02() {
         // TODO: compute the average age of all male
-        return 0L;
+        return EMPLOYEES
+                .stream()
+                .filter(employee -> employee.getGender() == Gender.MAN)
+//                .peek(employee -> System.out.println(employee))
+                .mapToInt(employee -> employee.getAge())
+//                .peek(age -> System.out.println(age))
+                .average().getAsDouble();
+
     }
 
     static long ex03() {
         // TODO: count how many employees are male aged 18 to 60 and women aged 18 to 55
-        return 0L;
+        return EMPLOYEES.stream()
+                .filter(employee -> employee.getAge() > 18)
```

Comment:
You need fix this condition to `employee.getAge() >= 18`

Reply:
Ow!!! Thank you very much!)))